src/Symfony/Component/Security/Core/User/LdapUserProvider.php

```diff
@@ -35,12 +35,16 @@ class LdapUserProvider implements UserProviderInterface
     private $defaultSearch;
     private $passwordAttribute;
 
-    public function __construct(LdapInterface $ldap, string $baseDn, string $searchDn = null, string $searchPassword = null, array $defaultRoles = array(), ?string $uidKey = 'sAMAccountName', string $filter = '({uid_key}={username})', string $passwordAttribute = null)
+    public function __construct(LdapInterface $ldap, string $baseDn, string $searchDn = null, string $searchPassword = null, array $defaultRoles = array(), string $uidKey = null, string $filter = null, string $passwordAttribute = null)
```

Comment:
I'm not sure what the collective opinion in the Symfony community is about this:
```php
args(string $uidKey = null, string $filter = null)
```
But in theory it has to be written like:
```php
args(string $uidKey = null, string $filter = null)
```
null is not an allowed type according to your signature, but it's listed as default value. In order to allow null to be a valid type, the signature should be:
```php
args(?string $uidKey = null, ?string $filter = null)
```
With this signature you allow null to be present. Sadly php accepts both due to 7.0 not having nullable types. 

I'm not sure if this should be fixed, what do you think @nicolas-grekas? Ideally I'd like to see this fix being applied on the whole codebase.

Reply:
To be fair, `array()` vs `[]` is just syntactic sugar, where as `string $foo = null` would actually be a type mismatch. But I understand the reasoning as of why you want to keep it out for now. I think with new features that do not require merging from 3.x, it should be fine though (so this change can stay without `?`).

Reply:
> would actually be a type mismatch

It's not, setting null as default value *means* the type is nullable. You're over-interpreting here.